tests/unit/test_json_schema_generator.py

```diff
@@ -0,0 +1,875 @@
+"""Tests for JSON Schema generation"""
```

Comment:
Many of the tests here were a bit hard to follow. There is room for improving clarity and expectations of what each test is setting up (Initial conditions), doing (Your actions), and accomplishing (Your results).

This kind of clarity is why I am a huge fan of following a GIVEN-WHEN-THEN structure, but we don't have to explicitly follow that.

Reply:
@BryanFauble I've added `GIVEN-WHEN-THEN` statements to the tests where it seem to make sense to add.